node/actors/bft/src/leader/tests.rs

```diff
@@ -1,39 +1,398 @@
-
-    consensus.leader.view = validator::ViewNumber(3);
-    consensus.leader.phase = validator::Phase::Commit;
-
-    let test_replica_msg =
-        consensus
-            .inner
-            .secret_key
-            .sign_msg(validator::ConsensusMsg::ReplicaCommit(
-                validator::ReplicaCommit {
-                    protocol_version: validator::CURRENT_VERSION,
-                    view: consensus.leader.view,
-                    proposal: rng.gen(),
-                },
-            ));
-
-    match consensus.leader.process_replica_commit(
-        ctx,
-        &consensus.inner,
-        test_replica_msg.cast().unwrap(),
-    ) {
-        Err(super::replica_commit::Error::UnexpectedProposal) => {}
-        res => panic!("unexpected result {res:?}"),
-    }
+//! ## Tests coverage
+//!
+//! - [x] replica_prepare_sanity
+//! - [x] replica_prepare_sanity_yield_leader_prepare
+//! - [x] replica_prepare_old_view
+//! - [x] replica_prepare_during_commit
+//! - [x] replica_prepare_not_leader_in_view
+//! - [x] replica_prepare_already_exists
+//! - [x] replica_prepare_num_received_below_threshold
+//! - [x] replica_prepare_invalid_sig
+//! - [x] replica_prepare_invalid_commit_qc
+//! - [x] replica_prepare_high_qc_of_current_view
+//! - [x] replica_prepare_high_qc_of_future_view
+//! - [x] replica_prepare_non_validator_signer // FAILS
+//!
+//! - [x] replica_commit_sanity
+//! - [x] replica_commit_sanity_yield_leader_commit
+//! - [x] replica_commit_old
+//! - [x] replica_commit_not_leader_in_view
+//! - [x] replica_commit_already_exists
+//! - [x] replica_commit_num_received_below_threshold
+//! - [x] replica_commit_invalid_sig
+//! - [x] replica_commit_unexpected_proposal
+//! - [x] replica_commit_protocol_version_mismatch // FAILS
+//!
+
+use assert_matches::assert_matches;
+use rand::Rng;
+
+use zksync_consensus_roles::validator::{
+    self, CommitQC, ConsensusMsg, LeaderCommit, LeaderPrepare, Phase, ProtocolVersion,
+    ReplicaCommit, ViewNumber,
+};
+
+use crate::testonly::ut_harness::UTHarness;
+
+use super::{
+    replica_commit::Error as ReplicaCommitError, replica_prepare::Error as ReplicaPrepareError,
+};
+
+#[tokio::test]
+async fn replica_prepare_sanity() {
+    let mut util = UTHarness::new().await;
+
+    let replica_prepare = util.new_current_replica_prepare(|_| {});
+    util.dispatch_replica_prepare(replica_prepare).unwrap();
+}
+
+#[tokio::test]
+async fn replica_prepare_sanity_yield_leader_prepare() {
+    let mut util = UTHarness::new().await;
+
+    let replica_prepare = util.new_current_replica_prepare(|_| {});
+    util.dispatch_replica_prepare(replica_prepare).unwrap();
+    util.recv_signed()
+        .await
+        .unwrap()
+        .cast::<LeaderPrepare>()
+        .unwrap();
+}
+
+#[tokio::test]
+async fn replica_prepare_old_view() {
+    let mut util = UTHarness::new().await;
+
+    util.set_replica_view(ViewNumber(1));
+    util.set_leader_view(ViewNumber(2));
+    util.set_leader_phase(Phase::Prepare);
```

Comment:
```suggestion
    util.set_leader_view(ViewNumber(2));
```
Unnecessary calls, I think.

Reply:
These are indeed the default value being set, but it might change (at least, the view number), and might not be self evident to everyone.